src/app/ui/recipe-details/recipe-details.component.ts

```diff
@@ -7,4 +11,12 @@ import { ChangeDetectionStrategy, Component } from '@angular/core';
   changeDetection: ChangeDetectionStrategy.OnPush
 })
 export class RecipeDetailsComponent {
+
+  recipe$: Observable<Recipe | null>
```

Comment:
```suggestion
  recipe$: Observable<Recipe | null> = this._route.data.pipe(
      map(({ recipe })=> recipe)
    )
```
We usually move simple logic from constructor / ngOnInit to property declaration.